mrchecker-framework-modules/mrchecker-core-module/src/main/java/com/capgemini/mrchecker/test/core/base/encryption/providers/DataEncryptionService.java

```diff
@@ -56,68 +54,63 @@ public static void delInstance() {
 		DataEncryptionService.instance = null;
 	}
 	
-	private String readSecret(File secretFile) {
-		try (Scanner scanner = new Scanner(secretFile)) {
-			return readLine(scanner, secretFile).trim();
-		} catch (FileNotFoundException e) {
-			throw new BFSecureModuleException("File does not exist: " + secretFile);
+	private String readSecret(InputStream secretSource) {
+		try (Scanner scanner = new Scanner(secretSource)) {
+			return readLine(scanner, secretSource).trim();
 		}
 	}
```

Comment:
I think a catch is missing here

Reply:
try-with-resources will autoclose scanner.